searx/engines/__init__.py

```diff
@@ -146,24 +128,27 @@ def load_engine(engine_data):
 
     engine.timeout += settings['outgoing'].get('extra_proxy_timeout', 0)
 
-    for category_name in engine.categories:
-        categories.setdefault(category_name, []).append(engine)
+    return engine
+
+
+def register_engine(engine):
+    if engine is None:
+        return
 
+    engines[engine.name] = engine
     if engine.shortcut in engine_shortcuts:
         logger.error('Engine config error: ambigious shortcut: {0}'.format(engine.shortcut))
         sys.exit(1)
-
     engine_shortcuts[engine.shortcut] = engine.name
-
-    return engine
+    for category_name in engine.categories:
+        categories.setdefault(category_name, []).append(engine)
 
 
 def load_engines(engine_list):
-    global engines, engine_shortcuts
```

Comment:
The references are kept, so there is no write access, so there is no need for `global`.